### ISSUE TITLE
Wrong self type only reported upon call if using VarArgs

### ISSUE THREAD

**Issue by troiganto** [python/mypy#16487]
**Bug Report**

Mypy does not raise an error if a method accepts nothing but a vararg list, but lacks a `self` parameter or `staticmethod` decorator.

This behavior is confusing because Mypy _does_ catch wrong self types for regular parameters.

**To Reproduce**

```python
class Class:
    def method(*args: int) -> int:
        return sum(args)

print(Class().method(1, 2, 3))
```

**Expected Behavior**

```
foo.py:2: error: Self argument missing for a non-static method (or an invalid type for self)  [misc]
```

**Actual Behavior**

```
foo.py:5: error: Invalid self argument "Class" to attribute function "method" with type "Callable[[VarArg(int)], Any]"  [misc]
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: 1.7.0
- Mypy command-line flags: `--strict`
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: 3.9
